frontend/packages/topology/src/layouts/ColaLayout.ts

```diff
@@ -532,7 +533,24 @@ class ColaLayout implements Layout {
     const newNodes: ColaNode[] = initialRun
       ? this.nodes
       : this.nodes.filter((node) => !this.nodesMap[node.element.getId()]);
-    const addingNodes = restart && newNodes.length > 0;
+    let addingNodes = restart && newNodes.length > 0;
+
+    if (restart && !addingNodes) {
+      this.groups.forEach((group) => {
+        const prevGroup = prevGroups.find((g) => g.element.getId() === group.element.getId());
+        if (!prevGroup) {
+          addingNodes = true;
+          newNodes.push(...group.leaves);
```

Comment:
This will mutate `this.nodes` when you're on your `initialRun`. Not sure if it matters, just figured I'd mention it.

Reply:
it's not obvious but initialRun and restart are never both true. I can update with a comment or a check. Thanks.